core/internal/pathtree/pathtree.go

```diff
@@ -34,8 +40,25 @@ type Format int
 const (
 	FormatYaml Format = iota
 	FormatJson
+	FormatJsonExt
 )
 
+// PathItem is a alternative representation of the item interface.
+//
+// The Value is a JSON string, which can be unmarshaled to any type.
```

Comment:
PathTree shouldn't have any JSON handling in it, that's a higher level concern. As I see it, what we need here is just the "tree with named nodes and arbitrary leaf values" data structure. The leaf value type can be generic and defined where PathTree is used.

Reply:
But it means we will need to traverse the tree at least twice

Reply:
i don't see why?

Reply:
ok, so new entry might have the json_string as another map if unmarshal it? and you need to add it as the relevant subtree in your whole tree... which means you need to traverse the tree again and add it

Reply:
I don't understand. If the JSON is handled completely outside of PathTree, there's no extra work within PathTree. Outside of PathTree, any JSON string has to be unmarshalled anyway; I don't see how this introduces any extra traversal.

Reply:
yeah, i think you right, we can do it outside and just provide pathitems!

Reply:
every place where we're now using the pathtree needs jsonification, so why not tie it with the data structure?

Reply:
see https://github.com/wandb/wandb/pull/7467#discussion_r1580234314

we shouldn't tie it with the data structure because it is orthogonal. Putting it all together would make it less clear that it's orthogonal *and* would allow the two concerns to get intertwined in the future, possibly by accident!